src/containers/Dashboard.vue

```diff
@@ -211,8 +210,10 @@ export default Vue.extend({
     listenerUnsubscriber();
   },
   methods: {
-    editSemesters(newSemesters: readonly FirestoreSemester[]) {
-      this.semesters = newSemesters;
+    editSemesters(
+      updater: (oldSemesters: readonly FirestoreSemester[]) => readonly FirestoreSemester[]
+    ) {
+      this.semesters = updater(this.semesters);
       this.recomputeRequirements();
     },
```

Comment:
Using an updater is necessary after switching to the new library, since it introduces some race conditions.

When you drag a course from A to B, it will remove the course from A and add it to B. A bad race condition happens when the course is first added to B, but the update of adding to B is not read before deleting course A, which causes the actions adding to B lost while the action of removing from A preserved. The result is that we see course is deleted after drag and drop.

I fixed the issue by using this updater function. The caller is forced to read from `oldSemester` from the updater function. Then at the root, we can always ensure that the updater is called with the latest data updated in root, so when adding/deleting things, child components will never read stale data. This pattern is common in React.